Feat/follow delete

queenriwon opened the pull request:
팔로우 기능에 이어 언팔로우 기능을 사용했습니다.
팔로우/언팔로우 코드 작성내용이 비슷하기 때문에 한 메서드에 토글방법을 사용하자는 의견이 나오긴 했지만, 나중에 맞팔로우 상태 설정이나, 친구기능으로의 확장성을 고려할 때 따로 메서드를 구현하는 것이 좋을 것 같다고 생각해 팔로우, 언팔로우 기능을 각각 구현했습니다.

조회의 경우 내가 팔로우 하고 있는 사람을 조회할 수 있으며, 내가 팔로우 한 유저가 얼마나 많은 팔로우를 받고 있는지(팔로잉) followCount까지 출력할 수 있습니다.
원래는 Member 엔티티에서 양방향으로 구현하려고 했다가, DB기능으로 갯수를 출력할 수 있다는 걸 알게되었습니다.
stream의 경우 조금 복잡해 보일 수 있겠으나,
Follow -> Member -> MemberListResponseDto로 변환하는 과정이라고 생각하시면 될 것 같습니다.

queenriwon committed:
Feat(follow): deleteFollow 언팔로우 기능 구현

follows 테이블에서 팔로우가 되어있는지 확인하고 언팔로우로 변경시키는 로직을 짜보았습니다.

queenriwon committed:
Feat(follow): findAllFollowerMembers 팔로우한 유저 조회

팔로우한 유저 리스트를 조회할 수 있습니다. 아직 팔로잉수는 0으로 작성되도록 했는데, 이부분 나중에 양방향을 사용할 수도 있을 것 같습니다.

queenriwon committed:
Merge branch 'dev' of https://github.com/teamnull0214/news-feed into feat/follow-delete

queenriwon committed:
Feat(member): findAllFollowerMembers 구현

로그인을 한 유저가 자신이 팔로우 한 유저들의 목록을 작성할 수 있게 작성하되, 팔로잉 수도 함께 출력되게 구현했습니다.
데이터 베이스 기본 기능을 사용하는 방법으로 양방향 연관관계 사용을 피했습니다.

queenriwon committed:
Merge branch 'dev' of https://github.com/teamnull0214/news-feed into feat/follow-delete

jiwonclvl commented on src/main/java/com/example/newsfeed/follow/service/FollowService.java:
```diff
@@ -40,4 +45,44 @@ public void createFollow(Long followerId, Long followingId) {
+                    return new MemberListResponseDto(member, followerCount);
+                })
+                .collect(Collectors.toList());
+    }
```
이 부분은 쿼리로 따로 빼지 않은 이유가 있나요? 궁금합니다!

hyeons22 commented:
DB기능 : 갯수출력 가능 <- 체크
확실히 스트림이 직관적이라 더 이해하기가 쉽네요
잘보고갑니다~

queenriwon committed:
Merge branch 'feat/post-like' of https://github.com/teamnull0214/news-feed into feat/follow-delete

queenriwon committed:
Merge branch 'dev' of https://github.com/teamnull0214/news-feed into feat/follow-delete

queenriwon committed:
Merge branch 'dev' of https://github.com/teamnull0214/news-feed into feat/follow-delete

queenriwon committed:
Refactor(follow): jpql을 사용하여 리팩토링

조회시 jpql을 사용하였고, memberId만으로 새로운 follow를 생성할 수 있도록 구현했습니다.
일부 postlike부분도 리팩토링했습니다.

queenriwon commented:
조회시 jpql을 사용하였고, memberId만으로 새로운 follow를 생성할 수 있도록 구현했습니다.
일부 postlike부분도 리팩토링했습니다.
